doc/lua_code_style_guidelines.md

```diff
@@ -0,0 +1,330 @@
+		another_function_call() then
+	do_something()
+end
+```
+
+* Function arguments are indented by two tabs if multiple arguments are in a line, same for definition continuations.
+```lua
+foo(bar, biz, "This is a long string..."
+		baz, qux, "Lua")
+
+function foo(a, b, c, d,
+		e, f, g, h)
+	[…]
+end
+```
+
+* If the function arguments contain a table, it's indented by one tab and if the arguments get own lines, it's indented like a table.
+```lua
+register_node("name", {
+	"This is a long string...",
+	0.3,
+})
+
+list = filterlist.create(
+	preparemodlist,
+	comparemod,
+	function()
+		return "no comma at the end"
+	end
+)
+```
+
+* When strings don't fit into the line, you should add the string (changes) to the next line(s) indented by one tab.
+```lua
+longvarname = longvarname ..
+	"Thanks for reading this example!"
+
+local retval =
+	"size[11.5,7.5,true]" ..
+	"label[0.5,0;" .. fgettext("World:") .. "]" ..
+	"label[1.75,0;" .. data.worldspec.name .. "]"
+```
+
+* When breaking around a binary operator you should break after the operator.
+```lua
+if a or b or c or d or
+		e or f then
+	foo()
+end
+```
+
+
+# Empty lines
+
+* Use a single empty line to separate sections of long functions.
+
+* Use two empty lines to separate top-level functions and large tables.
+
+* Do not use a empty line after conditional, looping, or function opening statements.
+
+<span style="color: #282;">Good:</span>
+```lua
+function foo()
+	if x then
+		bar()
+	end
+end
+```
+
+<span style="color: #b44;">Bad:</span>
+```lua
+function foo()
+
+	if x then
+
+		bar()
+	end
+end
+```
+
+* Don't leave white-space at the end of lines.
+
+# Spaces
+
+* Spaces are not used around parenthesis, brackets, or curly braces.
+
+<span style="color: #282;">Good:</span>
+```lua
+foo({baz=true})
+bar[1] = "Hello world"
+```
+
+<span style="color: #b44;">Bad:</span>
+```lua
+foo ( { baz=true } )
+bar [ 1 ]
+```
+
+* Spaces are used after, but not before, commas and semicolons.
+
+<span style="color: #282;">Good:</span>
+```lua
+foo(a, b, {c, d})
+```
+
+<span style="color: #b44;">Bad:</span>
+```lua
+foo(a,b,{c , d})
+```
+
+* Spaces are used around binary operators with following exceptions:
+    * There mustn't be spaces around the member access operator (".")
+    * Spaces around the concatenation operator ("..") are optional.
+    * In short one-line table definitions the spaces around the equals sign can be omitted.
+    * When in-/decrementing a variable by 1, the spaces around the + and - operator can be omitted if you want to "get the neighbour" of something, e.g. when increasing some counter variable.
+
+<span style="color: #282;">Good:</span>
+```lua
+local num = 2 * (3 / 4)
+foo({bar=true})
+foo({bar = true})
+local def = {
+	foo = true,
+	bar = false,
+}
+i = i+1
+sometable[#sometable+1] = v
+```
+
+<span style="color: #b44;">Bad:</span>
+```lua
+local num=2*(3/4)
+local def={
+	foo=true,
+	bar=false,
+}
+playerpos.y = playerpos.y+1  -- playerpos.y is not an integer
```

Comment:
what

Reply:
Idk why this comment is here. But removing it means changing content.